client/components/AvailableListsScreen.js

```diff
@@ -0,0 +1,77 @@
+import React from 'react';
+import Button from '@material-ui/core/Button';
+import Typography from '@material-ui/core/Typography';
+import CircularProgress from '@material-ui/core/CircularProgress';
+import {withRouter} from 'react-router';
+import {Link} from "react-router-dom";
+import {AddressInput} from "./AddressInput";
+import {getCoordinatesForAddress} from "../utils/geocoding";
+
+export const AvailableListsScreen = withRouter(class extends React.Component {
+    constructor() {
+        super();
+
+        this.state = {
+            address: null,
+            latitude: null,
+            longitude: null
+        };
+    }
+
+    async onSubmit() {
+        try {
+            await this.props.onSubmit(this.state.items, this.state.budget, this.state.address, this.state.latitude, this.state.longitude);
+            this.props.history.push('/');
+        } catch (e) {
+            console.error(e);
+        }
+    }
+
+    async onSelectAddress(address) {
+        let lat, lng;
+        const coordinates = await getCoordinatesForAddress(address);
+        if (coordinates){
+            lat = coordinates.lat;
+            lng = coordinates.lng;
+        }
+        this.setState({ 
+            address : address,
+            latitude : lat,
+            longitude : lng
+        });
+
+        await this.props.getLists(lat, lng);
+    }
+
+    render() {
+        if (!this.state.address) {
+            return (
+                <AddressInput onSelect={this.onSelectAddress.bind(this)}></AddressInput>
+            );
+        }
+
+        if (!this.props.lists || this.props.lists.length === 0){
+            return (
+                <div>No lists available in your area right now! Check back next time!</div>
+            )
+        }
+
+        if (Object.keys(this.props.lists).length === 0){
```

Comment:
Can `this.props.lists` ever be an object (besides an Array)?

Reply:
no this was old code from when i was troubleshooting that nested state thing, woops